api/types/patch.go

```diff
@@ -17,3 +17,12 @@ type Patch struct {
 	// Target points to the resources that the patch is applied to
 	Target *Selector `json:"target,omitempty" yaml:"target,omitempty"`
 }
+
+// Equals return true if p equals another.
+func (p *Patch) Equals(another Patch) bool {
+	targetEqual := (p.Target == another.Target) ||
+		(p.Target != nil && another.Target != nil && *p.Target == *another.Target)
+	return p.Path == another.Path &&
+		p.Patch == another.Patch &&
+		targetEqual
```

Comment:
minor nit - mind if you inlined the expression for targetEqual here instead of computing it first.  If target were an actual object we could have given it an Equals method. :)

Reply:
`Selector` itself can be compared by `==` directly since it only contains strings. The reason that I need to use `targetEqual` variable is `Target` is a pointer and it can be `nil`. Inline the expression make the return statement hard to read.

Reply:
akc, thanks